src/destreamer.ts

```diff
@@ -255,17 +256,23 @@ async function downloadVideo(videoUrls: string[], outputDirectories: string[], s
             }
         });
 
-        ffmpegCmd.on('error', (error: any) => {
-            cleanupFn();
-
-            console.log(`\nffmpeg returned an error: ${error.message}`);
-            process.exit(ERROR_CODE.UNK_FFMPEG_ERROR);
-        });
-
         process.on('SIGINT', cleanupFn);
 
         // let the magic begin...
         await new Promise((resolve: any, reject: any) => {
+            ffmpegCmd.on('error', (error: any) => {
+                if (argv.skip && error.message === `File '${outputPath}' already exists. Exiting.\n`) {
```

Comment:
I don't really like the string comparison (since it's a test condition against a particular implementation at one point in time) but i'm with you, i don't think we have a simple good alternative other than implement our own _does the video title match a file that already exist_ logic which we should probably avoid since FFmpeg does it anyway.

FFmpeg bails out with exit code 1 if destination already exists and we didn't pass `-n`, code 1 if we DID pass `-n`, also code 1 if sanity as we know it is burning to a crisp. Code 1.

<img src="https://user-images.githubusercontent.com/6472374/81110840-c7afc000-8f24-11ea-92ff-af36c627ea32.png" width="300">

My naive proposal would be to search for `exists` in error.message (`includes()`), not the whole thing. That has better chance to survive a rewrite of that error message in FFmpeg.

@kylon, is there mabe a Fessonia way to approach this? Went over the docs and there doesn't even seem to be a way to access the exit code from inside the `error` event handler.

Reply:
I would suggest to also check if the output path is included in the message, otherwise we would also accept some other error messages. I am confident that the file will always be in the output, since FFmpeg supports [multiple outputs](https://trac.ffmpeg.org/wiki/Creating%20multiple%20outputs) and the output would otherwise be quite useless.
```suggestion
                if (argv.skip && error.message.includes('exists') && error.message.includes(outputPath)) {
```